src/test/java/uk/gov/nationalarchives/pdi/step/jena/shacl/JenaShaclStepTest.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.nationalarchives.pdi.step.jena.shacl;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.riot.RDFDataMgr;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.pentaho.di.core.KettleEnvironment;
+import org.pentaho.di.core.RowMetaAndData;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.row.RowMeta;
+import org.pentaho.di.core.row.RowMetaInterface;
+import org.pentaho.di.core.row.value.ValueMetaSerializable;
+import org.pentaho.di.core.variables.Variables;
+import org.pentaho.di.trans.RowStepCollector;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.TransTestFactory;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class JenaShaclStepTest {
```

Comment:
I had previously described these tests that use TransTestFactory as Integration Tests as opposed to Unit Tests and ran them via the fail-safe plugin instead of the sure-fire plugin. However, I am not certain which they should be - let's have a discussion about that...

Reply:
Pentaho provide two examples for plugin tests at https://github.com/pentaho/pdi-sdk-plugins/blob/master/kettle-sdk-step-plugin/src/test/java/org/pentaho/di/sdk/samples/steps/demo/DemoStepTest.java (which is the one I based mine on) and https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/test/java/org/pentaho/di/trans/steps/rowgenerator/RowGeneratorUnitTest.java which I discovered later. It looks like the second one may be more of a true unit test (judging by the name at least) so I guess we should follow that approach. I originally attempted to create my own unit tests using mocks but I think plugin development was a bit of an afterthought for Pentaho and the API wasn't really created with this in mind - I found I was having to mock stuff that shouldn't be exposed through a true API (IMHO).